### PR TITLE
initial secretstore implementation of HashiCorp Vault

### PR DESCRIPTION
**Pull request by anubhavmishra**
# Description

This is an initial version of HashiCorp Vault secretstore implementation.

## Issue reference

This PR closes: #81 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: secretstores/vault/vault.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	}
+}
+
+// Init creates a HashiCorp Vault client
+func (v *vaultSecretStore) Init(metadata secretstores.Metadata) error {
+	props := metadata.Properties
+
+	// Get Vault address
+	address := props[componentVaultAddress]
+	if address == "" {
```
Comment (yaron2) [inline secretstores/vault/vault.go]: Can we make this a required property and add an error if not found or empty?

File: secretstores/vault/vault.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+		return secretstores.GetSecretResponse{Data: nil}, err
+	}
+
+	// Create get secret url
+	// TODO: Add support for versioned secrets when the secretstore request has support for it
+	vaultSecretPathAddr := fmt.Sprintf("%s/v1/secret/data/%s/%s?version=0", v.vaultAddress, v.vaultKVPrefix, req.Name)
+
+	httpReq, err := http.NewRequest(http.MethodGet, vaultSecretPathAddr, nil)
+	// Set vault token.
+	httpReq.Header.Set("X-Vault-Token", token)
```
Comment (youngbupark) [inline secretstores/vault/vault.go]: nit: it would be good to use const type for "X-Vault-Token"

File: secretstores/vault/vault.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+		io.Copy(&b, httpresp.Body)
+		return secretstores.GetSecretResponse{Data: nil}, fmt.Errorf("couldn't to get successful response: %#v, %s",
+			httpresp, b.String())
+	}
+
+	var d struct {
+		Data struct {
+			Data map[string]string `json:"data"`
+		} `json:"data"`
+	}
```
Comment (youngbupark) [inline secretstores/vault/vault.go]: Would you mind defining this struct with type explicitly?

File: secretstores/vault/vault.go (hunk 1 above)
Comment (anubhavmishra) [reply secretstores/vault/vault.go]: @yaron2 we default to the local address when not found.

```
if address == "" {
	v.vaultAddress = defaultVaultAddress
}
```

Comment (yaron2) [reply secretstores/vault/vault.go]: Ok

**Comment by anubhavmishra** [comment]
@yaron2 @youngbupark I added a few tests and resolved your comments.

**Comment by yaron2** [comment]
LGTM, thank you!

**Pull request merged**